homestead.root/usr/share/clearwater/bin/poll_cx.py

```diff
@@ -0,0 +1,287 @@
+    _log.debug("Total UAR client errors = {}".format(total_client_errors))
+    _log.debug("Total UAR server errors = {}".format(total_server_errors))
+
+    return (total_successes, total_client_errors, total_server_errors)
+
+# Count all the successes, client errors and server errors for the SAR table
+#
+# Returns the counts in a tuple
+def SAR_responses():
+    # Determine the head of the segment of the table we are going to read
+    sar_segment_head = table_segment_head % 11
+
+    # Get the successes and client and server errors for the BASE responses and
+    # add them to the successes and client and server errors for the _3GPP responses
+    (total_successes, total_client_errors, total_server_errors) = \
+        map(operator.add, count_responses(sar_segment_head + ".0", [2001], []), \
+                          count_responses(sar_segment_head + ".1", [], [5001]))
+
+    # Add the count of timeouts to the server error count
+    total_server_errors += count_timeouts(sar_segment_head)
+
+    _log.debug("Total SAR successes = {}".format(total_successes))
+    _log.debug("Total SAR client errors = {}".format(total_client_errors))
+    _log.debug("Total SAR server errors = {}".format(total_server_errors))
+
+    return (total_successes, total_client_errors, total_server_errors)
+
+# Count all the successes, client errors and server errors for the MAR table
+#
+# Returns the counts in a tuple
+def MAR_responses():
+    # Determine the head of the segment of the table we are going to read
+    mar_segment_head = table_segment_head % 10
+
+    # Get the successes and client and server errors for the BASE responses and
+    # add them to the successes and client and server errors for the _3GPP responses
+    (total_successes, total_client_errors, total_server_errors) = \
+        map(operator.add, count_responses(mar_segment_head + ".0", [2001], []), \
+                          count_responses(mar_segment_head + ".1", [], [5001]))
+
+    # Add the count of timeouts to the server error count
+    total_server_errors += count_timeouts(mar_segment_head)
+
+    _log.debug("Total MAR successes = {}".format(total_successes))
+    _log.debug("Total MAR client errors = {}".format(total_client_errors))
+    _log.debug("Total MAR server errors = {}".format(total_server_errors))
+
+    return (total_successes, total_client_errors, total_server_errors)
+
+# Perform a single check of the statistics to determine whether the Diameter
+# stack might be broken.
+#
+# Note that this is called twice to ensure that we do not make a false
+# positive determination due to "tick rollover"
+def are_all_requests_failing():
+
+    # Collate all "Previous 5 minute" statistics for the UAR, MAR, SAR and LIR
+    # statistic tables
+    (total_successes, total_client_errors, total_server_errors) = \
+              map(operator.add, map(operator.add, MAR_responses(), SAR_responses()), \
+                                map(operator.add, UAR_responses(), LIR_responses()))
```

Comment:
Actually, now that I've got rid of the command specific collators, map(operator.add makes more sense (because I'm just calling the same routine repeatedly maintaining a running count)